api/envoy/config/filter/http/cache/v2/cache.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.cache.v2;
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.http.cache.v2";
+option java_outer_classname = "CacheProto";
+option java_multiple_files = true;
+
+// [#protodoc-title: HTTP Cache Filter]
+// Pluggable HTTP Cache :ref:`overview <config_http_filters_cache>`.
+// [#extension: envoy.filters.http.cache]
+message Cache {
+  // List of query params.
+  message QueryParams {
+    repeated string query_string_keys = 1;
+  }
+
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
+
+    // If true, exclude the host from the cache key. Set to true if your origins' responses don't
+    // ever depend on host.
+    bool exclude_host = 2;
+
+    oneof query_string_keys {
+      // If set, only query params in this list are included in the cache
+      // key. Any other query params will not affect cache lookup. If set to an
+      // empty list, the query string will be entirely ignored in cache lookups.
+      QueryParams query_string_keys_included = 3;
+
+      // If set, query params in this list are excluded from the cache key, and
+      // will not affect cache lookup. Setting this to an empty list is
+      // equivalent to not setting it (but is less efficient).
+      QueryParams query_string_keys_excluded = 4;
```

Comment:
Might there be a need to have both a whitelist and blacklist here?

Reply:
They are currently formulated as matching the whole query param name, with no wildcard/regex/prefix/etc matching. As such, it's hard to see any value in having both a whitelist and a blacklist at the same time.

If these were to become regexes or similar, then having both a whitelist and a blacklist at the same time would be useful. So far, I haven't run across anyone needing this soft of pattern matching, but it could happen.

Reply:
My main concern is a perennial Envoy API one; every filter/extension likes to cook up its own matching scheme, even for similar concerns. `RouteConfiguration` already has matchers for query parameters. Could these be reused? I would rather we maintain a consistent method for specifying matchers across the API where possible.

Reply:
Excellent suggestion--thanks! Done.